lib/zlib.js

```diff
@@ -686,57 +686,78 @@ Zlib.prototype.params = function params(level, strategy, callback) {
 // generic zlib
 // minimal 2-byte header
 function Deflate(opts) {
-  if (!(this instanceof Deflate))
+  if (!(this instanceof Deflate)) {
+    process.emitWarning(`Instantiating Deflate class without 'new' is deprecated.`,
+                        'DeprecationWarning', 'DEP0184');
```

Comment:
FWIW when https://github.com/nodejs/node/pull/54869 lands there will be a helper utility for this warning

Reply:
Now that that PR has landed, I recommend using the new utility